models/auth/oauth2.go

```diff
@@ -31,6 +31,8 @@ type OAuth2Application struct {
 	Name         string
 	ClientID     string `xorm:"unique"`
 	ClientSecret string
+	// https://datatracker.ietf.org/doc/html/rfc6749#section-2.1
+	Confidential bool               `xorm:"NOT NULL DEFAULT TRUE"`
```

Comment:
According to the RFC: OAuth defines two client types. So to match the document, could the field be "ClientType=confidential" or "ClientType=public" ?

Reply:
I thought bool simplest to represent two choices. GitLab uses a similar Confidential option https://github.com/go-gitea/gitea/issues/21299#issuecomment-1262021100